src/components/Common/NewDelegationAccordion/NewDelegationAccordion.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  Accordion,
+  AccordionHeader,
+  AccordionContent,
+  ButtonVariant,
+  ButtonColor,
+  Button,
+} from '@altinn/altinn-design-system';
+import { useState } from 'react';
+import { t } from 'i18next';
+
+import type { DelegableOrg } from '@/rtk/features/delegableOrg/delegableOrgSlice';
+import type { DelegableApi } from '@/rtk/features/delegableApi/delegableApiSlice';
+import { ReactComponent as MinusCircle } from '@/assets/MinusCircle.svg';
+import { ReactComponent as AddCircle } from '@/assets/AddCircle.svg';
+
+import classes from './NewDelegationAccordion.module.css';
+
+export enum NewDelegationAccordionButtonType {
+  Add = 'add',
+  Remove = 'remove',
+}
+
+export interface NewDelegationAccordionProps {
+  title: string;
+  subtitle: string;
+  hasOrgNr?: boolean;
+  description: string;
+  buttonType: NewDelegationAccordionButtonType;
+  callback: () => { payload: DelegableOrg | DelegableApi; type: string };
```

Comment:
Another thing here: Does the component need to know what the payload of the callback is here? The parameters aren't used here, they're baked into the callback outside of the component.

So can't this callback-parameter be more openly defined here anyway?

Reply:
If we use any we will receive lint error and not using anything gives typescript error. I also think defining expected payload-object gives value. I don't know what you mean by callback-parameter being more openly defined.

Reply:
I mean that you can do like [here](https://github.com/Altinn/altinn-access-management-frontend/blob/025ecc80b05e785b8017c698b2895a62d9573023/src/components/ApiDelegationOverviewPage/OrgDelegationOverviewPageContent/OrgDelegationAccordion/DeletableListItem/DeletableListItem.tsx#L12)

NewDelegationAccordion does not inherently need to be related to Orgs or Apis, and it doesn't need to know whether or not your callback is related to DelegableOrg or DelegableApi or what you do with them. It just needs to call the function you give it.

Reply:
At least that's how I see it.
I don't know what value these payload-objects would give here

Reply:
You're right, I forgot it could be done by returning void.